vignettes/download_github_comments.Rmd

```diff
@@ -27,40 +36,41 @@ require(magrittr)
-
-While the above is true for **comments**, the first message of every issue and every pull request (which also include a title) is not regarded by GitHub as a comment. For example, suppose one issue was opened by Author A, and contain only one reply by Author B. In this case, A and B established communication, and we would like this interaction to be reflected in the final table. However, if we were to only use the **comments** endpoint, we would only obtain Author's B comment, and not Author's A. The same is true in Pull Requests.
-
-Therefore, in this Notebook we have to rely on three endpoints from the GitHub API: The `Issue endpoint` to obtain the "first comment" of every issue, the `Pull Request endpoint` to obtain the "first comment" of every pull request, then finally the `Issue and Pull Request Comment endpoint`, which provides comments for both issue and pull requests together.
-
+To use the pipeline, you must specify the organization and project of interest, and your token.
 
-# Project Configuration File
-
-To use the pipeline, you must specify the organization and project of interest, and your token. Obtain a github token following the instructions [here](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token).
-
-```{r}
+```{r warning=FALSE}
 conf <- parse_config("../conf/kaiaulu.yml")
+owner <- get_github_owner(conf, "project_key_1") # Has to match github organization (e.g. github.com/sailuh)
+repo <- get_github_repo(conf, "project_key_1") # Has to match github repository (e.g. github.com/sailuh/perceive)
 
+# Path you wish to save all raw data.
 save_path_issue_refresh <- get_github_issue_search_path(conf, "project_key_1")
 save_path_issue <- get_github_issue_path(conf, "project_key_1")
 save_path_pull_request <- get_github_pull_request_path(conf, "project_key_1")
+save_path_pr_comments <- get_github_pr_comments_path(conf, "project_key_1")
 save_path_issue_or_pr_comments <- get_github_issue_or_pr_comment_path(conf, "project_key_1")
 save_path_commit <- get_github_commit_path(conf, "project_key_1")
-# Path you wish to save all raw data. A folder with the repo name and sub-folders will be created.
+# Create all folder directories
+create_file_directory(conf)
 
-owner <- get_github_owner(conf, "project_key_1") # Has to match github organization (e.g. github.com/sailuh)
-repo <- get_github_repo(conf, "project_key_1") # Has to match github repository (e.g. github.com/sailuh/perceive)
 # your file github_token (a text file) contains the GitHub token API
```

Comment:
I would add more details on this. On my device, the .ssh folder did not exist, nor did the github_tokens folder (so I had to create one). Also, the file extension of github_tokens needs to be removed, so that it is a Document type (usually defaults to .txt or .rtf). I think this might be worth explaining.

Reply:
Ignore this if it is not relevant to you: I noticed that the getter functions are creating directories, and I think that is from your work on making directories if they do not exist? If so, I do not think you need output for folders that already exist:
```
Filepath: ../../rawdata/kaiaulu/github/sailuh_kaiaulu/issue_search/ already exists.
Filepath: ../../rawdata/kaiaulu/github/sailuh_kaiaulu/issue_event/ already exists.
```
Output for directory not existing and being created would be good though.